src/uu/tail/src/args.rs

```diff
@@ -208,6 +220,62 @@ impl Settings {
             .map(|v| v.map(|string| Input::from(string.clone())).collect())
             .unwrap_or_default();
 
+        if !inputs.is_empty() {
+            const DEFAULT_MODE: char = 'l';
+            const DEFAULT_INTVAL: u64 = 10;
+            let mut name = inputs[0].display_name.to_string();
+            let mut default_mode = false;
+            if let Some(mut mode) = name.chars().last() {
```

Comment:
maybe move most this into a separate function

Reply:
moved to FilterMode impl, I guess it fits better there